File: HW04/131044061_CSE241_HW04/Circle.cpp
#include<string>
#include<fstream>
#include <iostream>
#include "Circle.h"
using namespace std;

Circle::Circle() {
	radius = 0;
	pos_x = 0;
	pos_y = 0;

}

void Circle::set_radius(int r) {
	if(r>1){
		radius = r;
	}
	else{
		cout<<"circ olusmadi"<<endl;
	}
}

int Circle::get_radius() const {
	return radius;
}

void Circle::set_coords(double p_x, double p_y) {
	if(p_x >= radius && p_y >= radius){
		pos_x = p_x;
		pos_y = p_y;
	}
	else{
		cout<<"circ koordinant hatasi"<<endl;
	}
}

double Circle::get_pos_x() const {
	return pos_x;
}

double Circle::get_pos_y() const {
	return pos_y;
}

const double Circle::area() {
	return (3.1415) * (double)get_radius() * (double)get_radius();
}

const double Circle::lenght() {
	return 2 * (3.1415) * (double)get_radius();
}

const Circle Circle::operator ++() {
	this->pos_x++;
	this->pos_y++;
	this->set_coords(pos_x, pos_y);
	Circle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const Circle Circle::operator ++(int pre) {
	++this->pos_x;
	++this->pos_y;
	this->set_coords(pos_x, pos_y);
	Circle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const Circle Circle::operator --() {
	this->pos_x--;
	this->pos_y--;
	this->set_coords(pos_x, pos_y);
	Circle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const Circle Circle::operator --(int pre) {
	--this->pos_x;
	--this->pos_y;
	this->set_coords(pos_x, pos_y);
	Circle temp;
	temp.set_coords(pos_x, pos_y);
	return temp;
}

const bool Circle::operator ==(Circle& other) {
	return this->area() == other.area();
}

const bool Circle::operator !=(Circle& other) {
	return !(this->area() == other.area());
}

int Circle::get_numOfShape() {
	numOfShape++;
	return numOfShape;
}

double Circle::result_area(Circle& circ) {
	return numOfShape * circ.area();
}

const Circle Circle::operator +(double num) {
	Circle temp;
	temp.set_radius(this->get_radius() + num);
	return temp;
}

const Circle Circle::operator -(double num) {
	Circle temp;
	temp.set_radius(this->get_radius() + num);
	return temp;
}

ofstream& operator <<(ofstream& ofs, const Circle& circ) {
	ofs<<circ.get_idntfy()<<" "<<circ.get_coord_x()<<circ.get_pos_x()<<"' "<<circ.get_coord_y()
			<<circ.get_pos_y()<<"' "<<circ.get_rad()<<circ.get_radius()<<"' "<<circ.get_stroke()
			<<circ.fill<<endl;
	return ofs;
}

string Circle::get_idntfy() const {
	return idntfy;
}

string Circle::get_coord_x() const {
	return coord_x;
}

string Circle::get_coord_y() const {
	return coord_y;
}

string Circle::get_rad() const {
	return rad;
}

string Circle::get_stroke() const {
	return stroke;
}

void Circle::draw(ofstream & ofs) const {
	ofs<<*this;
}

int Circle::reset_numOfShape() {
	numOfShape = 0;
	return numOfShape;
}

File: HW07/131044061_CSE241_HW07_caner_karakas/hw07/src/Circle.java
import java.awt.Color;
import java.awt.Graphics;
import java.lang.*;

import javax.swing.JPanel;

public class Circle extends JPanel implements Shape {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 6168523935497005135L;
	private double radius;
	private static int numOfShape = 0;
	private double pos_x, pos_y;
	/**
	 * empty constructor
	 */
	public Circle() {}
	/**
	 * takeCircleObject
	 * @param c
	 * @throws Exception
	 */
	public Circle(Circle c) throws Exception {
		this.set_radius(c.get_radius());
		this.set_coords(c.get_pos_x(), c.get_pos_y());
	}
	/**
	 * take radius
	 * @param r
	 * @throws Exception
	 */
	public Circle(double r) throws Exception {
		set_radius(r);
	}

	@Override
	public int compareTo(Shape other) {
		if(this.area() == other.area())
			return 1;
		return 0;
	}
	public void set_radius(double r) throws Exception {
		if(r>0)
			radius = r;
		else {
			new Exception ();
		}
	}
	@Override
	public double area() {return (Math.PI) * radius * radius;}

	@Override
	public double perimeter() {return 2 * (Math.PI) * radius;}

	@Override
	public Shape increment() {
		++pos_x;
		++pos_y;
		return null;
	}
	/**
	 * set_coords
	 * @param x
	 * @param y
	 * @throws Exception
	 */
	void set_coords(double x, double y) throws Exception{
		if(x >= radius && y >= radius){
			set_pos_x(x);
			set_pos_y(y);
		}
		else{
			/*System.out.printf("Error!!");
			System.exit(0);*/
			new Exception ();
		}
	}
	
	@Override
	public void paintComponent(Graphics graph){
		super.paintComponent(graph);
		graph.setColor(Color.GREEN);
		graph.drawOval((int)this.get_pos_x()-(int)this.get_radius(), (int)this.get_pos_y()-(int)this.get_radius(), (int)this.get_radius()*2, (int)this.get_radius()*2);
	}

	@Override
	public Shape decrement() {
		--pos_x;
		--pos_y;
		return this;
	}

	@Override
	public void draw(Graphics graph) {
		this.paintComponent(graph);
	}
	
	public double get_radius() {return radius;}
	
	static int get_numOfShape(){numOfShape++;return numOfShape;}
	
	static int reset_numOfShape(){numOfShape = 0;return numOfShape;}
	
	static double result_area(Circle circ){return numOfShape * circ.area();}

	public double get_pos_x() {return pos_x;}

	public void set_pos_x(double x) {this.pos_x = x;}

	public double get_pos_y() {return pos_y;}

	public void set_pos_y(double y) {this.pos_y = y;}

}
